example android, gradle and maven projects

lugehorsam opened the pull request:
I thought it'd be a good idea to add some example projects that I've been using to test sdk integration into this repository.

novabyte commented:
@lugehorsam A great idea to add some example folder structures for different Java ecosystem build tools. A few small nits about the code:

- I've not come across `_` in Java package names before. Is this common usage or should we opt for the best practice around package name conventions?
- I think we should avoid the blocking version of the Future `.get()` in examples because it pauses the main thread and is not the style we'd recommend in production apps.
- The various code needs the usual Apache-2 license headers please.

lugehorsam commented:
@novabyte ready for re-review.

Not sure why Android studio added the `_` but I cleaned that up. I also added the more real-world future example -- is that something we'd like in our documentation rather than the blocking `.get()` call? Thinking of the code blocks in our docs.

Also documenting the change I made yet again to the Maven coordinates: Separate artifact IDs is necessary because otherwise the shadow jar pom.xml nukes the dependency list for the regular jar pom.xml. That is why https://forum.heroiclabs.com/t/android-studio/1023 was reported in the forum. Once you create separate ids, Jitpack has an inefficiency that appends the artifact id to the group name, resulting in some redundancy. There are open issues on this (e.g., https://github.com/jitpack/jitpack.io/issues/209)

novabyte commented:
Nice one @lugehorsam 👍
